tests/main/views/test_marketplace.py

```diff
@@ -34,6 +35,10 @@ def test_data_analytics_track_page_view_is_shown_if_account_created_flag_flash_m
         data = res.get_data(as_text=True)
 
         assert 'data-analytics="trackPageView" data-url="buyers?account-created=true"' in data
+        # however this should not be shown as a regular flash message
+        flash_banner_match = re.search('<p class="banner-message">\s*(.*)', data, re.MULTILINE)
```

Comment:
No particular reason, I suppose there's an increasing order of complexity from 

 - just checking if some text is in the response (i.e. what the assert just before the line I added does)
 - using a regex to check if some text is in there, and extract some nearby text too (i.e. this thing)
 - using xpath to build a full DOM and do stuff with that.

They are all fine (and maybe my assumption that regex's are cheaper than xpath is wrong), but choosing the least complex option in each case seems okay. (Although consistency is good too, so in a test that already used all xpath, I'd probably have gone with that.)